amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -952,26 +953,20 @@ protected ShardConsumer buildConsumer(@NonNull final ShardInfo shardInfo,
                                           @NonNull final LeaseCleanupManager leaseCleanupManager) {
         ShardRecordProcessorCheckpointer checkpointer = coordinatorConfig.coordinatorFactory().createRecordProcessorCheckpointer(shardInfo,
                         checkpoint);
-        // The only case where streamName is not available will be when multistreamtracker not set. In this case,
-        // get the default stream name for the single stream application.
-        final StreamIdentifier streamIdentifier = getStreamIdentifier(shardInfo.streamIdentifierSerOpt());
-
-        // Irrespective of single stream app or multi stream app, streamConfig should always be available.
-        // If we have a shardInfo, that is not present in currentStreamConfigMap for whatever reason, then return default stream config
-        // to gracefully complete the reading.
-        StreamConfig streamConfig = currentStreamConfigMap.get(streamIdentifier);
-        if (streamConfig == null) {
-            streamConfig = streamTracker.createStreamConfig(streamIdentifier);
+        final StreamConfig streamConfig = shardInfo.streamConfig();
+        if (!currentStreamConfigMap.containsKey(streamConfig.streamIdentifier())) {
             log.info("Created orphan {}", streamConfig);
         }
-        Validate.notNull(streamConfig, "StreamConfig should not be null");
+        /*
+         * NOTE: RecordsPublisher#createGetRecordsCache(ShardInfo, StreamConfig, MetricsFactory) is deprecated.
+         *  RecordsPublisher#createGetRecordsCache(ShardInfo, MetricsFactory) will be called directly in the future.
+         */
         RecordsPublisher cache = retrievalConfig.retrievalFactory().createGetRecordsCache(shardInfo, streamConfig, metricsFactory);
```

Comment:
Is the reason we are not changing this now is that there could be some clients which pass in their own interface of RetrievalFactory and may not implement the new function yet? Wouldn't that still mean its backwards incompatible if we ever change its, so it would have to be a major version upgrade?